docs/enhancements/tenancy-based-auth.md

```diff
@@ -0,0 +1,103 @@
+
+Filtering for you own clusters:
+The list-clusters API shall have an additional query parameter that filters the clusters by owners only:
+`GET /v2/clusters?owneronly=true`
+
+## Authorization Flow Overview
+![authorization flow](./auth_flow.png)
+
+Following is a brief overview of the authorization layer of the assisted installer and its interaction with AMS.
+
+There are three actors in the flow:
+- An authenticated client (User or Agent) carrying a bearer or access token that identifies it.
+- Assisted Service security middleware, acting as a reverse proxy, which receives a request to access a resource and works together with AMS to establish the right of the client to access it.
+- Account Management Service (AMS) that holds the OCM authorization details for a user.
+
+Upon receiving a request from a client, the authorization middleware issues access_review and capability_review requests to AMS to establish the user/agent's static role:
+
+
+| Role          | Description                                | Criteria (AMS)                                     |
+|---------------|--------------------------------------------|----------------------------------------------------|
+| ReadOnlyAdmin | Customer support personas                  | Has bare_metal_installer_admin capability          |
+| User          | Users/Agents allowed to create AI clusters | Has create permission on BareMetalCluster resource |
+
+After the static roles are established, the authorization layer verifies that the requested API is allowed for that particular role. For example, A support user (with AdminReadOnly role) can only view a cluster and download its logs but not modify it.
+
+If a call is issued by a `User` there is an additional authorization phase where the middleware consults the assisted service database and verifies that the user has ownership over the requested resource.
+
+In the current implementation, `ownership` means that the user created the resource (cluster or infraEnv). The enhancement suggests a more elaborate way to establish ownership, taking into account tenancy as well as dynamic roles.
+
+### AMS / Assisted Service Integration
+AMS access authorization API is based upon rules that govern the access rights of users to subscriptions. Subscriptions bear a close connection with OCP clusters. 
+
+Assisted-Service, on the other hand, identifies resources by their local identifiers. There are two kinds of resources in Assisted Service:
+- A resource belonging to a Cluster.
+- A resource belonging to an InfraEnv.
+
+The association between those two systems is done on a cluster level. A cluster record in assisted service keeps track of its corresponding subscription ID in AMS. 
+
+InfraEnv has no parallel association. Therefore, any integration with AMS security API for InfraEnv resources should be either done via the bound cluster's subscription id or fall back to the local rules of access.
```

Comment:
What do you mean by "the local rules of access"?

Reply:
>InfraEnv has no parallel association. Therefore, any integration with AMS security API for InfraEnv resources should be either done via the bound cluster's subscription id or fall back to the local rules of access.

The usage of the words "should be" implies that this is a suggestion, but since we're not in the "suggested changes" part yet, It's a bit not clear whether this is a description of the current state of things or a suggestion

Reply:
clarified